internal/pubsub/driver/driver.go

```diff
@@ -74,9 +74,15 @@ type Topic interface {
 // Subscription receives published messages.
 type Subscription interface {
 	// ReceiveBatch should return a batch of messages that have queued up
-	// for the subscription on the server. If no messages are available
-	// yet, it must block until there is at least one, or the context is
-	// done.
+	// for the subscription on the server.
+	//
+	// If there is a transient failure, this method should not retry but
+	// should return a nil slice and an error. The concrete API will take
+	// care of retry logic.
+	//
+	// If the service returns no messages for some other reason, this
```

Comment:
OK, so the concrete implementation will loop? But then it's important that this method waits for a while before it returns no messages. To be specific, a GCP implementation that set `ReturnImmediately` to true or false would both satisfy this requirement, but the first would cause busy-waiting.